compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1175,51 +1175,314 @@ OMR::ARM64::TreeEvaluator::vpermEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+ * @param[in] value: immediate value
+ *
+ * @return instruction cursor if an instruction is successfully generated and otherwise returns NULL
+ */
+static
+TR::Instruction *tryToGenerateMovImm16ShiftedInstruction(TR::Node *node, TR::CodeGenerator *cg, TR::Register *treg, uint16_t value)
+   {
+   uint8_t lower8bit = value & 0xff;
+   uint8_t upper8bit = (value >> 8) & 0xff;
+   if (lower8bit == upper8bit)
+      {
+      return generateTrg1ImmInstruction(cg, TR::InstOpCode::vmovi16b, node, treg, lower8bit);
+      }
+
+   TR::Instruction *instr;
+   if (instr = tryToGenerateImm16ShiftedInstrucion(node, cg, treg, TR::InstOpCode::vmovi8h, value))
+      {
+      return instr;
+      }
+   else if (instr = tryToGenerateImm16ShiftedInstrucion(node, cg, treg, TR::InstOpCode::vmvni8h, ~value))
+      {
+      return instr;
+      }
+   return NULL;
+   }
+
+/**
+ * @brief Tries to generate a SIMD move immediate instruction for 32bit value.
+ *
+ * @param[in] node: node
+ * @param[in] cg: CodeGenerator
+ * @param[in] treg: target register
+ * @param[in] value: immediate value
+ *
+ * @return instruction cursor if an instruction is successfully generated and otherwise returns NULL
+ */
+static
+TR::Instruction *tryToGenerateMovImm32ShiftedInstruction(TR::Node *node, TR::CodeGenerator *cg, TR::Register *treg, uint32_t value)
+   {
+   uint16_t lower16bit = value & 0xffff;
+   uint16_t upper16bit = (value >> 16) & 0xffff;
+   TR::Instruction *instr;
+
+   if (lower16bit == upper16bit)
+      {
+      if (instr = tryToGenerateMovImm16ShiftedInstruction(node, cg, treg, lower16bit));
+         {
+         return instr;
+         }
+      }
+   else if (instr = tryToGenerateImm32ShiftedInstruction(node, cg, treg, TR::InstOpCode::vmovi4s, value))
+      {
+      return instr;
+      }
+   else if (instr = tryToGenerateImm32ShiftedInstruction(node, cg, treg, TR::InstOpCode::vmvni4s, ~value))
+      {
+      return instr;
+      }
+   else if (instr = tryToGenerateImm32ShiftingOnesInstruction(node, cg, treg, TR::InstOpCode::vmovi4s_one, value))
+      {
+      return instr;
+      }
+   else if (instr = tryToGenerateImm32ShiftingOnesInstruction(node, cg, treg, TR::InstOpCode::vmvni4s_one, ~value))
+      {
+      return instr;
+      }
+   return NULL;
+   }
+
+TR::Instruction *
+OMR::ARM64::TreeEvaluator::vsplatsImmediateHelper(TR::Node *node, TR::CodeGenerator *cg, TR::Node *firstChild, TR::DataType elementType, TR::Register *treg)
+   {
+   if (firstChild->getOpCode().isLoadConst())
+      {
+      auto constValue = firstChild->getConstValue();
+      switch (elementType)
+         {
+         case TR::Int8:
+            return generateTrg1ImmInstruction(cg, TR::InstOpCode::vmovi16b, node, treg, constValue & 0xff);
+
+         case TR::Int16:
+            {
+            uint16_t value = static_cast<uint16_t>(constValue & 0xffff);
+            TR::Instruction *instr = tryToGenerateMovImm16ShiftedInstruction(node, cg, treg, value);
+            if (instr)
```

Comment:
How about adding an empty line before `if (instr)` in the same way as the `Int32`/`Int64` cases.

Reply:
Added a blank line.